File: leet/remove_nth_node_from_end_of_list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        
        
        if head is None:
            return none
        
        
        length = 0
        
        dummy = head
        
        while dummy is not None:
            dummy = dummy.next
            length+=1
            
        if length == n:
            return head.next
        
        temp = head
        count = 1
        
        while count < length-n:
            temp = temp.next
            count+=1
         
        temp.next = temp.next.next
        return head

File: leet/remove_nth_node_from_end_of_list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        
        ListNode temp = head;
        
        int length = getLength(head);
        
        int count = 1;
        
        if(length-n == 0){
            return head.next;
        }
        
        while (count < (length-n)){
            temp = temp.next;
            count++;
        }
        
        temp.next = temp.next.next;
        return head;
        
    
    }
    
    public int getLength(ListNode head){
        if(head == null){
            return 0;
        }
        
        ListNode dummy = head;
        
        int count = 0;
        while(dummy != null){
            dummy = dummy.next;
            count++;
        }
        
        return count;
    }


}